src/dune_engine/source_tree.ml

```diff
@@ -121,6 +121,81 @@ let filter_source_files =
   in
   ref (fun _ -> Memo.Build.return (fun _dir fn -> not (is_temp_file fn)))
 
+(* CR-someday amokhov: It would be nice to derive this instead. For now let's
+   trust I haven't messed this up. *)
+let unix_error_equal (x : Unix.error) (y : Unix.error) =
```

Comment:
Shouldn't we move this function to its own file (maybe `otherlibs/stdune-unstable/unix_error.ml`)?

Another suggestion is that a future-proof equality function should be written like this:

```
| E2BIG, E2BIG -> true
| E2BIG, _ | _, E2BIG -> false
...
```

then we don't need the catch-all clause and readers don't have to wonder if there are some cases missing or not.

Reply:
Good idea, will do that!